### PR TITLE
Travis-CI improvements: cache .m2 and apt status; separate unit tests & pipeline tests; test -runLocal

### PR DESCRIPTION
**Pull request by jakebiesinger**
@JavierJia @anbangx @Nan-Zhang 

Our builds have started to time out, thanks to a loooong `mvn` download process in some cases and the fact that our unittests take a long time to run.  I've added caching to the `~/.m2` directory to speed up subsequent builds and have broken up the run into 3 steps-- unittests, pipeline tests, and pipelinetests with `-runLocal`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+cache:
+    - apt
+    - $HOME/.m2
 
+# build matrix
 jdk:
     - oraclejdk7
     - openjdk7
+env:
+    - UNITTESTS=true PIPELINE_TESTS=false RUN_LOCAL=""
```
Comment (JavierJia) [inline .travis.yml]: So this env means travis will run each configuration ? Awesome ~

Comment (jakebiesinger) [inline .travis.yml]: Yep.  So there are currently 6 total tests to run-- 2 JDK's x 3 environment variable settings.  The run_tests.sh script de-multiplexes them and only runs the requested tests.

**Comment by jakebiesinger** [comment]
Forgive the force-push.  Had a spelling error on `UNIT_TESTS`

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,19 @@
+# configuration
 language: java
+cache:
+    - apt
+    - $HOME/.m2
```
Comment (JavierJia) [inline .travis.yml:5]: So adding the `cache` statement will not clear the existing folder ? 
What's the meaning of `apt` here ?

Comment (jakebiesinger) [inline .travis.yml:5]: `cache`d folders are actually stored in S3, as I understand it, and restored on a per-branch basis (so every branch gets its own version of the cache).  For us, it looks like the folders are exactly as they were on the last build.

The `apt` one is a travis special flag meaning "keep track of any changes made by apt-get"-- specifically, it should only have to install `graphviz` once if this works.

**Comment by JavierJia** [comment]
LGTM :+1:

**Comment by jakebiesinger** [comment]
Any other comments, @Nan-Zhang or @anbangx ?

**Comment by anbangx** [comment]
LGTM

On Mon, Nov 4, 2013 at 8:08 PM, Jake Biesinger notifications@github.comwrote:

> Any other comments, @Nan-Zhang https://github.com/Nan-Zhang or @anbangxhttps://github.com/anbangx?
> 
> ?
> Reply to this email directly or view it on GitHubhttps://github.com/uci-cbcl/genomix/pull/52#issuecomment-27746374
> .

## 

Best Regards,

Anbang Xu

**Comment by jakebiesinger** [comment]
Okay, @Nan-Zhang I'm merging without you. Hope that's okay! :wink:

**Pull request merged**